terror/terror.go

```diff
@@ -119,23 +128,38 @@ func (ec ErrClass) NotEqualClass(err error) bool {
 	return !ec.EqualClass(err)
 }
 
-// New defines an *Error with an error code and an error message.
-// Usually used to create base *Error.
-// Attention:
-// this method is not goroutine-safe and
-// usually be used in global variable initializer
-func (ec ErrClass) New(code ErrCode, message string) *Error {
+func (ec ErrClass) initError(code ErrCode) string {
 	clsMap, ok := ErrClassToMySQLCodes[ec]
 	if !ok {
 		clsMap = make(map[ErrCode]struct{})
 		ErrClassToMySQLCodes[ec] = clsMap
 	}
 	clsMap[code] = struct{}{}
-	return &Error{
-		class:   ec,
-		code:    code,
-		message: message,
-	}
+	class := errClass2Desc[ec]
+	rfcCode := fmt.Sprintf("%s:%d", class, code)
+	rfcCode2errClass[class] = ec
+	return rfcCode
+}
+
+// New defines an *Error with an error code and an error message.
+// Usually used to create base *Error.
+// Attention:
+// this method is not goroutine-safe and
+// usually be used in global variable initializer
+func (ec ErrClass) New(code ErrCode, message string) *Error {
+	rfcCode := ec.initError(code)
+	err := errors.Normalize(message, errors.MySQLErrorCode(int(code)), errors.RFCCodeText(rfcCode))
+	errCodeMap[code] = err
```

Comment:
Is there any concurrent problem?

Reply:
yes, this method is not groutine-safe, but always used for global init.